x/authz/types/msgs.go

```diff
@@ -0,0 +1,247 @@
+		return sdkerrors.Wrap(ErrInvalidGranter, "missing granter address")
+	}
+	if msg.Grantee == "" {
+		return sdkerrors.Wrap(ErrInvalidGrantee, "missing grantee address")
+	}
+	if msg.Expiration.Unix() < time.Now().Unix() {
+		return sdkerrors.Wrap(ErrInvalidExpirationTime, "Time can't be in the past")
+	}
+
+	return nil
+}
+
+func (msg *MsgGrantAuthorization) GetAuthorization() Authorization {
+	authorization, ok := msg.Authorization.GetCachedValue().(Authorization)
+	if !ok {
+		return nil
+	}
+	return authorization
+}
+
+func (msg *MsgGrantAuthorization) SetAuthorization(authorization Authorization) error {
+	m, ok := authorization.(proto.Message)
+	if !ok {
+		return fmt.Errorf("can't proto marshal %T", m)
+	}
+	any, err := types.NewAnyWithValue(m)
+	if err != nil {
+		return err
+	}
+	msg.Authorization = any
+	return nil
+}
+
+func (msg MsgExecAuthorized) UnpackInterfaces(unpacker types.AnyUnpacker) error {
+	for _, x := range msg.Msgs {
+		var msgExecAuthorized sdk.MsgRequest
+		err := unpacker.UnpackAny(x, &msgExecAuthorized)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// UnpackInterfaces implements UnpackInterfacesMessage.UnpackInterfaces
+func (msg MsgGrantAuthorization) UnpackInterfaces(unpacker types.AnyUnpacker) error {
+	var authorization Authorization
+	return unpacker.UnpackAny(msg.Authorization, &authorization)
+}
+
+// String implements the Stringer interface
+func (msg MsgGrantAuthorization) String() string {
+	out, _ := yaml.Marshal(msg)
+	return string(out)
+}
+
+// NewMsgRevokeAuthorization creates a new MsgRevokeAuthorization
+//nolint:interfacer
+func NewMsgRevokeAuthorization(granter sdk.AccAddress, grantee sdk.AccAddress, authorizationMsgType string) MsgRevokeAuthorization {
+	return MsgRevokeAuthorization{
+		Granter:              granter.String(),
+		Grantee:              grantee.String(),
+		AuthorizationMsgType: authorizationMsgType,
+	}
+}
+
+// Route implements Msg
+func (msg MsgRevokeAuthorization) Route() string { return RouterKey }
+
+// Type implements Msg
+func (msg MsgRevokeAuthorization) Type() string { return TypeMsgRevokeAuthorization }
+
+// GetSigners implements Msg
+func (msg MsgRevokeAuthorization) GetSigners() []sdk.AccAddress {
+	granter, err := sdk.AccAddressFromBech32(msg.Granter)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{granter}
+}
+
+// GetSignBytes implements Msg
+func (msg MsgRevokeAuthorization) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(&msg)
+	return sdk.MustSortJSON(bz)
+}
+
+// ValidateBasic implements Msg
+func (msg MsgRevokeAuthorization) ValidateBasic() error {
+	if msg.Granter == "" {
+		return sdkerrors.Wrap(ErrInvalidGranter, "missing granter address")
+	}
+	if msg.Grantee == "" {
+		return sdkerrors.Wrap(ErrInvalidGrantee, "missing grantee address")
+	}
+	return nil
+}
+
+// String implements the Stringer interface
+func (msg MsgRevokeAuthorization) String() string {
+	out, _ := yaml.Marshal(msg)
+	return string(out)
+}
```

Comment:
```suggestion
```

These `String` methods feel unnecessary. Let's just use the default proto-generated `String` methods.